docs/architecture/MLOps_process.md

```diff
@@ -0,0 +1,207 @@
+
+Let's discuss both of them.
+
+### CI/CD on the development branch ###
+
+This approach is most common in the case we can prepare all required artifacts in advance such as models and libraries, store them somewhere, and mark them as **production ready**.
+
+For example, we run training from the development branch on the full dataset to produce a model. The model is evaluated and approved. We can then mark the model using a custom attribute such as a  **production** tag. The CD pipeline will use the **latest** model with the **production** tag. In this case code in the development branch is not used as-is in production, it's used to do training in the development environment. So it's OK to use just one branch, for example, **main**, as the primary source for development and run CI/CD.
+
+Rollback can be implemented by removing the **production** tag from the latest model and execute the CI/CD again to pick up the previous version ready for production.
+
+![CICD](../media/mlops_arch_cicd.png)
+
+### Multi-branch strategy ###
+
+You can also use several branches to move your code from development to production. This strategy works well if you deploy the code itself, don’t update the production environment often, and have several versions of the production environment, for different partners or departments for example.
+
+In the previous sections we have introduced two kind of branches: feature branches and the development branches. Both of them are designed to support development rather than production, and we are building MLOps processes around these branches. Even though the development branch is stable, we cannot move it to production as-is. For example, we might have updated training pipelines, but no model exists based on it yet. Another example, somebody could activate a wrong model to be used in scoring service that is not critical for development branch, but critical for production. We propose to move development branch to production in two stages:
+
+* once the development branch is ready for production, move code alongside with the current model to a QA branch. This branch is going to be used for testing to double check that the code and the models work fine. Additionally, we can test our deployment scripts here.
+* once all tests on the QA branch have been completed successfully, we can move the code and the models from the QA branch to the production branch.
+
+The reason to use branches rather than tags is ability to execute some additional DevOps pipelines prior to commit code from one branch to another. In these pipelines we can implement approval process as well as move ML pipelines between environments.
+
+The PR Build on the development branch to QA branch should include the following tasks:
+
+* Deploy only what's needed for production. In most cases you don’t need to deploy training pipelines in production environment, only the scoring infrastructure.
```

Comment:
Should we have a section for scenarios where training needs to be done on production since in development/QA we have no access to production data?